### PR TITLE
Ensure correct items re-deployed

### LINKED ISSUE #55: Enable re-deploy of unhealthy pods even when not changed
At the moment, if a pod fails (e.g. no image etc...) a subsequent deploy will pass as the failed resource is not re-deployed.

Either:
1. Add a function to determine health of an existing pod or controller (using existing code) to determine if a deploy needs to run when md5 matches.
2. Delete the md5 label for failed resources at deploy time (I think I prefer this).

### PR DESCRIPTION
**Pull request by lewismarshall**
We need to do two things:
1. Re-deploy failing pods (Fixes #55).
2. Redeploy pods or controllers with changing secrets (Fixes by #57).

### CODE REVIEW

**Comment by jp-raju** [comment]
Hi Lewis,
Hope you are doing well.
I am keep getting these updates from all the team members.Would it be possible to remove me from the update list? Is it something I can do myself?
Regards,JP
Date: Wed, 16 Dec 2015 02:09:48 -0800

**Comment by lewismarshall** [comment]
Tested and working with NFIDD and FDCS...

Can be tested with tag v0.6.4-rc4

**Comment by lewismarshall** [comment]
@vaijab It'll be good to get this into tools...

File: libs/kb8_volumes.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      if volume.has_key?('secret')
+        @secrets << volume['secret']['secretName']
+      end
+    end
+  end
+
+  def uses_secret?(secret_name)
+    @secrets.include?(secret_name)
+  end
+end
```
Comment (vaijab) [inline libs/kb8_volumes.rb:22]: Missing new line

**Pull request merged**